lib/icingadb/icingadbchecktask.cpp

```diff
@@ -0,0 +1,398 @@
+	} else {
+		msgbuf << "never";
+	}
+
+	checkGreater(syncAgo, syncAgoThresholds);
+
+	if (syncWhen) {
+		msgbuf << "\n* Last sync took: " << syncTook << " seconds";
+		checkGreater(syncTook, syncTookThresholds);
+		perfdata->Add(new PerfdataValue("sync_took", syncTook, false, "seconds", syncTookThresholds.Warning, syncTookThresholds.Critical, 0));
+	}
+
+	perfdata->Add(new PerfdataValue("queries", qps, false, "", Empty, Empty, 0));
+	perfdata->Add(new PerfdataValue("queries_1min", redis->GetQueryCount(60), Empty, Empty, 0));
+	perfdata->Add(new PerfdataValue("queries_5mins", redis->GetQueryCount(5 * 60), Empty, Empty, 0));
+	perfdata->Add(new PerfdataValue("queries_15mins", redis->GetQueryCount(15 * 60), Empty, Empty, 0));
+	perfdata->Add(new PerfdataValue("pending_queries", pendingQueries, false, "", pendingQueriesThresholds.Warning, pendingQueriesThresholds.Critical, 0));
+	perfdata->Add(new PerfdataValue("down_for", downFor, false, "seconds", downForThresholds.Warning, downForThresholds.Critical, 0));
+	perfdata->Add(new PerfdataValue("heartbeat_lag", heartbeatLag, false, "seconds", heartbeatThresholds.Warning, heartbeatThresholds.Critical));
+	perfdata->Add(new PerfdataValue("idle_for", idleFor, false, "seconds", idleForThresholds.Warning, idleForThresholds.Critical));
+	perfdata->Add(new PerfdataValue("history_backlog", historyBacklog, false, "seconds", historyBacklogThresholds.Warning, historyBacklogThresholds.Critical, 0));
+
+	for (auto& kv : statsPerOp) {
+		auto perMin (kv.second.UpdateAndGetValues(now, 60));
+		auto perSec (perMin / 60.0);
+		auto thresholds (thresholdsByOp.find(kv.first));
+
+		msgbuf << "\n* " << perSec << " " << kv.first << "/s";
+
+		if (thresholds != thresholdsByOp.end()) {
+			checkLower(perSec, thresholds->second);
+		}
+
+		perfdata->Add(new PerfdataValue(kv.first, perSec, false, "", Empty, Empty, 0));
+		perfdata->Add(new PerfdataValue(kv.first + "_1min", perMin, false, "", Empty, Empty, 0));
+		perfdata->Add(new PerfdataValue(kv.first + "_5mins", kv.second.UpdateAndGetValues(now, 5 * 60), false, "", Empty, Empty, 0));
+		perfdata->Add(new PerfdataValue(kv.first + "_15mins", kv.second.UpdateAndGetValues(now, 15 * 60), false, "", Empty, Empty, 0));
+	}
+
+	{
+		static boost::regex wellNamedUnits (":(bytes|seconds)$");
+		ObjectLock lock (goMetricsByCumulativity);
+
+		for (auto& kv : goMetricsByCumulativity) {
+			bool cumulative = kv.first == "cumulative";
+			Dictionary::Ptr goMetricsPerCumulativity = kv.second;
+			ObjectLock lock (goMetricsPerCumulativity);
+
+			for (auto& kv : goMetricsPerCumulativity) {
+				std::string unit;
+				boost::smatch what;
+
+				if (boost::regex_search(kv.first.GetData(), what, wellNamedUnits)) {
+					unit = what[1];
+				}
+
+				bool counter = cumulative && unit.empty();
+
+				perfdata->Add(new PerfdataValue(kv.first, kv.second, counter, std::move(unit)));
```

Comment:
These names contain `/`, `:`, and `-`, which at least is inconsistent and I'm not sure whether this might cause trouble with some writers down the road.

Reply:
These comply to the Nagios guidelines and our graphers have been built to handle such.